src/Symfony/Component/PropertyAccess/Tests/PropertyAccessorTest.php

```diff
@@ -148,6 +148,8 @@ public function testGetValueNotModifyObjectException()
 
         try {
             $propertyAccessor->getValue($object, 'firstName[1]');
+
+            $this->fail('Expected an exception to be thrown when accessing an invalid element.');
```

Comment:
If you remove the try catch, you could make a separate test with an `@expectedException for this case`